python/ray/serve/api.py

```diff
@@ -1758,13 +1777,13 @@ def run(
     # TODO (jiaodong): Resolve circular reference in pipeline codebase and serve
     from ray.serve.pipeline.api import build as pipeline_build
 
-    client = start(detached=True, http_options={"host": host, "port": port})
-
+    if isinstance(target, Application):
```

Comment:
Technically, this isn't fully implemented in this PR but reverting it to the PR in Master will break some tests because of all the merging. Any thoughts on what to do? I'm ok with keeping this semi-implementation for now.

Reply:
let's leave the semi-impl and fix merge conflicts whoever gets their in first...